### PR TITLE
Creating token with ingress host info when using nodeport on OpenShift

### PR DESCRIPTION
**Pull request by fgiorgetti**

### CODE REVIEW

File: client/connector_token_create.go [hunk 1]
```diff
@@ -99,67 +99,68 @@ func configureHostPorts(result *RouterHostPorts, cli *VanClient, namespace strin
 	if namespace == "" {
 		namespace = cli.Namespace
 	}
-	ok, err := configureHostPortsFromRoutes(result, cli, namespace)
```
Comment (grs) [inline client/connector_token_create.go:102]: I *believe* that the logic for cert tokens already checks whether routes are actually in use. The problem is with claim tokens.

File: client/token_claim_create.go [hunk 2]
```diff
@@ -73,26 +73,26 @@ func (cli *VanClient) TokenClaimCreate(ctx context.Context, name string, passwor
 	}
 	host := fmt.Sprintf("%s.%s", types.ControllerServiceName, cli.Namespace)
 	localOnly := true
-	if cli.RouteClient != nil {
-		route, err := cli.RouteClient.Routes(cli.Namespace).Get(types.ClaimRedemptionRouteName, metav1.GetOptions{})
```
Comment (grs) [inline client/token_claim_create.go:77]: Another approach - the one used for creating cert tokens - is to check the error and to treat IsNotFound as implying routes are not in use.

File: client/token_claim_create.go [hunk 3]
```diff
@@ -73,26 +73,26 @@ func (cli *VanClient) TokenClaimCreate(ctx context.Context, name string, passwor
...
+		localOnly = false
+	} else if cli.RouteClient != nil {
+		route, err := cli.RouteClient.Routes(cli.Namespace).Get(types.ClaimRedemptionRouteName, metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
+		host = route.Spec.Host
+		port = 443
+		localOnly = false
+	} else if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
```
Comment (grs) [inline client/token_claim_create.go:94]: Will this not result in the same error if ingress=loadbalancer on openshift? (As previous else if clause will be true).

File: client/connector_token_create.go (hunk 1 above)
Comment (fgiorgetti) [reply client/connector_token_create.go:102]: Correct, it is not returning an error when not found occurs. I will revert it.

File: client/token_claim_create.go (hunk 2 above)
Comment (fgiorgetti) [reply client/token_claim_create.go:77]: I like that approach and will make it similar here as well.

File: client/token_claim_create.go (hunk 3 above)
Comment (fgiorgetti) [reply client/token_claim_create.go:94]: Yes, the approach you recommended is supposed to fix both cases.

**Review by grs — APPROVED** [review]

**Comment by fgiorgetti** [comment]
This resolves #551

**Pull request merged**